packages/graz/src/provider/events.tsx

```diff
@@ -23,10 +24,45 @@ import { WalletType } from "../types/wallet";
 export const useGrazEvents = () => {
   const isSessionActive =
     typeof window !== "undefined" && window.sessionStorage.getItem(RECONNECT_SESSION_KEY) === "Active";
-  const { _reconnect, _onReconnectFailed, _reconnectConnector } = useGrazInternalStore();
+  const { _reconnect, _onReconnectFailed, _reconnectConnector, iframeOptions, chains } = useGrazInternalStore();
   const { activeChainIds: activeChains, wcSignClients } = useGrazSessionStore();
   const isReconnectConnectorReady = checkWallet(_reconnectConnector || undefined);
 
+  // Auto connect to iframe if possible.
+  useEffect(() => {
+    if (
+      !iframeOptions ||
+      iframeOptions.autoConnect === null ||
+      !iframeOptions.allowedIframeParentOrigins.length ||
+      !chains
+    ) {
+      return;
+    }
+
+    const autoConnectChains = [iframeOptions.autoConnect || []].flat();
+    // Select auto-connect chains from provided chains.
+    const chainIds = autoConnectChains.length
+      ? chains.flatMap((c) => (autoConnectChains.includes(c.chainId) ? c.chainId : []))
+      : // If no auto-connect chains, attempt connection to all provided.
+        chains.map((c) => c.chainId);
```

Comment:
I prefer it's not auto connect when there's no auto-connect chains rather than connect to all provided. it will more make sense since `iframeOptions.autoConnect` is a `ChainId` not a boolean

```suggestion
    const chainIds = chains.flatMap((c) => (autoConnectChains.includes(c.chainId) ? c.chainId : []))
```